cmd/agent/common/autodiscovery.go

```diff
@@ -72,28 +71,16 @@ func setupAutoDiscovery(confSearchPaths []string, metaScheduler *scheduler.MetaS
 			}
 		}
 
-		ccaInAD := util.CcaInAD()
-		if ccaInAD {
-			var enableContainerProvider bool
-			for _, p := range incompatibleProviders {
-				if _, found := uniqueConfigProviders[p]; found {
-					enableContainerProvider = true
-					delete(uniqueConfigProviders, p)
-				}
+		var enableContainerProvider bool
+		for _, p := range incompatibleProviders {
```

Comment:
Don't recall why we called it `incompatibleProviders`, but now it's clear it could be better named `legacyProviders`, since they're only listed to provide backwards compatibility with older configs now.